### PR TITLE
Fix AutoLogin `printf` and `>` issues

### LINKED ISSUE #654: AutoLogin not working
## Describe the bug
Using SDDM and DWM does not configure autologin and I suspect the script doesn't work on many systems as it does not check for `autologin` group, nor does it add the group, or assign the group to the user. 

## Expected behavior
Auto Login

## Screenshots
image

## Checklist
- [x] I checked for duplicate issues.
- [x] I checked already existing discussions.
- [x] This issue is not included in the roadmap.
- [x] This issue is present on both stable and development branches.

### PR DESCRIPTION
**Pull request by adamperkowski**
## Type of Change
- [x] Bug fix
- [x] Refactoring
- [x] Hotfix

## Description
A LOT of bugs here. Someone (@guruswarupa) didn't test this.

## Issues / other PRs related

- Resolves #654

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/utils/auto-login.sh [hunk 1]
```diff
@@ -43,9 +43,9 @@ configure_lightdm() {
     printf "%b" "Enter username for LightDM autologin: "
     read -r user
 
-    "$ESCALATION_TOOL" "printf '[Seat:*]' > /etc/lightdm/lightdm.conf.d/50-autologin.conf"
-    "$ESCALATION_TOOL" "printf 'autologin-user=$user' >> /etc/lightdm/lightdm.conf.d/50-autologin.conf"
-    "$ESCALATION_TOOL" "printf 'autologin-user-timeout=0' >> /etc/lightdm/lightdm.conf.d/50-autologin.conf"
+    printf "%b\n" '[Seat:*]' | "$ESCALATION_TOOL" tee /etc/lightdm/lightdm.conf.d/50-autologin.conf
+    printf "%s\n" "autologin-user=$user" | "$ESCALATION_TOOL" tee -a /etc/lightdm/lightdm.conf.d/50-autologin.conf
+    printf "%b\n" 'autologin-user-timeout=0' | "$ESCALATION_TOOL" tee -a /etc/lightdm/lightdm.conf.d/50-autologin.conf
```
Comment (nnyyxxxx) [inline core/tabs/utils/auto-login.sh:48]: 50-autologin.conf is wrong

File: core/tabs/utils/auto-login.sh [hunk 2]
```diff
@@ -63,9 +63,9 @@ configure_gdm() {
     printf "%b" "Enter username for GDM autologin: "
     read -r user
 
-    "$ESCALATION_TOOL" "printf '[daemon]' > /etc/gdm/custom.conf"
-    "$ESCALATION_TOOL" "printf 'AutomaticLoginEnable = true' >> /etc/gdm/custom.conf"
-    "$ESCALATION_TOOL" "printf 'AutomaticLogin = $user' >> /etc/gdm/custom.conf"
+    printf "%b\n" '[daemon]' | "$ESCALATION_TOOL" tee /etc/gdm/custom.conf
+    printf "%b\n" 'AutomaticLoginEnable = true' | "$ESCALATION_TOOL" tee -a /etc/gdm/custom.conf
```
Comment (nnyyxxxx) [inline core/tabs/utils/auto-login.sh:67]: custom.conf is also wrong

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

**Comment by nnyyxxxx** [comment]
@adamperkowski you'll have to go through and find the correct paths for each one, some of these are wrong

File: core/tabs/utils/auto-login.sh (hunk 1 above)
Comment (nnyyxxxx) [reply core/tabs/utils/auto-login.sh:48]: lightdm.conf.d is not even a subdir

**Comment by adamperkowski** [comment]
@nnyyxxxx Who did this, I'd like to have a friendly talk with them...

**Comment by nnyyxxxx** [comment]
> @nnyyxxxx Who did this? I'd like to have a friendly talk with them...

guru

**Comment by adamperkowski** [comment]
@guruswarupa ...

File: core/tabs/utils/auto-login.sh (hunk 2 above)
Comment (adamperkowski) [reply core/tabs/utils/auto-login.sh:67]: This actually is right according to the [Archwiki](https://wiki.archlinux.org/title/GDM#Automatic_login).

**Comment by adamperkowski** [comment]
https://wiki.archlinux.org/title/LightDM#Enabling_autologin

**Review by nnyyxxxx — APPROVED** [review]

File: core/tabs/utils/auto-login.sh [hunk 3]
```diff
@@ -43,17 +43,18 @@ configure_lightdm() {
...
 
     printf "%b\n" "LightDM has been configured for autologin."
 }
 
 # Function to remove LightDM autologin
 remove_lightdm_autologin() {
     printf "%b\n" "Removing LightDM autologin configuration..."
-    "$ESCALATION_TOOL" rm -f /etc/lightdm/lightdm.conf.d/50-autologin.conf
+    "$ESCALATION_TOOL" sed -i '/^\[Seat:\*]/d' /etc/lightdm/lightdm.conf
+    "$ESCALATION_TOOL" sed -i '/^autologin-/d' /etc/lightdm/lightdm.conf
```
Comment (cartercanedy) [inline core/tabs/utils/auto-login.sh:57]: Change to `"$ESCALATION_TOOL" sed -i''...` for compat with bsd sed

**Review by cartercanedy — CHANGES_REQUESTED** [review]

**Comment by cartercanedy** [comment]
I'm pretty sure that the empty `-i` arg to I needs to be joined like `sed -i'' ...`, just based off of my own usage and the man pages

**Comment by nnyyxxxx** [comment]
> I was wrong
> 
> The bsd man pages are so much better than the ones for gnutils

wrong

**Comment by nnyyxxxx** [comment]
thats only for bsd

**Comment by cartercanedy** [comment]
> thats only for bsd

? 

Which form are you talking about

**Comment by nnyyxxxx** [comment]
> > thats only for bsd
> 
> ?
> 
> Which form are you talking about

the empty arg u mentioned in your edited message, only works for linux, bsd on the otherhand it would not work on

**Comment by cartercanedy** [comment]
> > > thats only for bsd
> > 
> > ?
> > 
> > Which form are you talking about
> 
> the empty arg u mentioned in your edited message, only works for linux, bsd on the otherhand it would not work on

`sed -i'.old' 's/foo/bar/' test.txt` produces a file named foo.txt.old

**Comment by nnyyxxxx** [comment]
> sed -i'.old' 's/foo/bar/' test.txt

that **will** work on bsd, but if u take the argument away it **wont**

**Comment by nnyyxxxx** [comment]
if there is no argument on linux it will work though, so you were only half right.

**Comment by cartercanedy** [comment]
> > sed -i'.old' 's/foo/bar/' test.txt
> 
> that **will** work on bsd, but if u take the argument away it **wont**

Correct, that's why I suggested the change

**Comment by cartercanedy** [comment]
I want saying that I was wrong to suggest the change, just that the empty arg needed to be concatenated to the flag @nnyyxxxx

**Comment by nnyyxxxx** [comment]
> I wasn't saying that I was wrong to suggest the change, just that the empty arg needed to be concatenated to the flag @nnyyxxxx

i understand i was only saying that you were partially wrong

**Comment by cartercanedy** [comment]
Ok, glad we agree...
Right?

**Review by cartercanedy — APPROVED** [review]
:)

**Comment by adamperkowski** [comment]
Co-authors for you :)

**Comment by guruswarupa** [comment]
> @guruswarupa ...

It seemed to work when those lines which you changed were echo. Later while changing it to printf, it broke. I did test on my system before adding it.

**Comment by cartercanedy** [comment]
> It seemed to work when those lines which you changed were echo. Later while changing it to printf, it broke. I did test on my system before adding it.

On its face that's incorrect. The single string argument being passed to sudo *will* fail because it will assume that it's looking for an executable named `echo '...' > ...`, which fails regardless of whether you substitute echo for printf. You'd either have to run `sudo bash -c "..."` or do what @adamperkowski did and utilize `sudo tee` to get the permissions right

**Comment by nnyyxxxx** [comment]
> > It seemed to work when those lines which you changed were echo. Later while changing it to printf, it broke. I did test on my system before adding it.
> 
> On its face that's incorrect. The single string argument being passed to sudo _will_ fail because it will assume that it's looking for an executable named `"echo '...' > ..."`, which fails regardless of whether you substitute echo for printf. You'd either have to run `sudo bash -c "..."` or do what @adamperkowski did and utilize `sudo tee` to get the permissions right

i'd recommend against using sub shells

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**